### PR TITLE
Logic for converting Financial Year into Calendar year for RPD data request - API

### PR DESCRIPTION
**Pull request by sunitaguptaEviden**
Logic for converting Financial Year into Calendar year for RPD data request - API

### CODE REVIEW

File: src/EPR.Calculator.API/Utils/FinancialYear.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        /// <summary>
+        /// Extracts the first year from a financial year string in the format "YYYY-YY".
+        /// </summary>
+        /// <param name="value">The financial year string to parse, in the format "YYYY-YY".</param>
+        /// <returns>The first year as a string.</returns>
+        /// <exception cref="FormatException">Thrown when the format is invalid.</exception>
+        public static string FinancialYearAsString(string value)
+        {
+            var years = value.Split('-');
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Utils/FinancialYear.cs:45]: What happens if value is null or Empty

File: src/EPR.Calculator.API/Utils/FinancialYear.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            if (years.Length != 2 || !int.TryParse(years[0], out int year) || years[0].Length != 4 || years[1].Length != 2)
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Utils/FinancialYear.cs:46]: Please use Regex Pattern

File: src/EPR.Calculator.API/Utils/FinancialYear.cs (hunk 1 above)
Comment (sunitaguptaEviden) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs]: @udaydefra Added null check - Done

File: src/EPR.Calculator.API/Utils/FinancialYear.cs (hunk 2 above)
Comment (sunitaguptaEviden) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs]: @udaydefra Replace with Regex Pattern - Done

**Review by johnbenjamin-defra — APPROVED** [review]

File: src/EPR.Calculator.API/Utils/FinancialYear.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        /// Extracts the first year from a financial year string in the format "YYYY-YY".
+        /// </summary>
+        /// <param name="value">The financial year string to parse, in the format "YYYY-YY".</param>
+        /// <returns>The first year as a string.</returns>
+        /// <exception cref="FormatException">Thrown when the format is invalid.</exception>
+        public static string FinancialYearAsString(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                throw new ArgumentException("Financial year cannot be null or empty", nameof(value));
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Utils/FinancialYear.cs:48]: Should this be ArgumentNullException?

File: src/EPR.Calculator.API/Utils/FinancialYear.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        /// Converts a financial year string to the previous calendar year as a string.
+        /// </summary>
+        /// <param name="value">The financial year string to convert, in the format "YYYY-YY".</param>
+        /// <returns>The previous calendar year as a string.</returns>
+        /// <exception cref="ArgumentException">Thrown when the financial year string is null or empty.</exception>
+        public static string ToCalendarYearAsString(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                throw new ArgumentException("Financial year cannot be null or empty", nameof(value));
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Utils/FinancialYear.cs:72]: Should this be ArgumentNullException?

File: src/EPR.Calculator.API/Utils/FinancialYear.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        /// <summary>
+        /// Gets the financial year.
+        /// </summary>
+        public DateTime Value { get; } = value;
+
+        /// <summary>
+        /// Converts the financial year to the previous calendar year.
+        /// </summary>
+        /// <returns>The previous calendar year as a <see cref="DateTime"/>.</returns>
+        public DateTime ToCalendarYear() => new(this.value.Year - 1, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Utils/FinancialYear.cs:17]: Not sure what this logic is? Can we discuss

File: src/EPR.Calculator.API/Utils/FinancialYear.cs (hunk 4 above)
Comment (sunitaguptaEviden) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs:72]: @udaydefra ArgumentNullException is specifically for null arguments whereas 
ArgumentException can be used to indicate that an argument is invalid for reasons other than being null, such as an empty string or containing only whitespace, so ArgumentException is the valid exception

File: src/EPR.Calculator.API/Utils/FinancialYear.cs (hunk 3 above)
Comment (udaydefra) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs:48]: I am ok with this

File: src/EPR.Calculator.API/Utils/FinancialYear.cs (hunk 5 above)
Comment (udaydefra) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs:17]: Why are we using UTC?

Comment (sunitaguptaEviden) [reply src/EPR.Calculator.API/Utils/FinancialYear.cs:17]: @udaydefra - This was the fix used for one of the error given by Sonar Qube in the service project and I kept it same in API as well, if you want I can remove from API project?

**Review by udaydefra — APPROVED** [review]

**Review by johnbenjamin-defra — APPROVED** [review]

**Pull request merged**